### PR TITLE
[JENKINS-24671] Add username and referrer to slow-requests

### PR DESCRIPTION
**Pull request by christ66**
This pull request adds the username of the person experiencing slow http requests. It also adds the referrer header in the event a plugin invokes the api and is unaware of it.

@reviewbybees

### CODE REVIEW

**Comment by alecharp** [comment]
:+1:

File: src/main/java/com/cloudbees/jenkins/support/slowrequest/InflightRequest.java [hunk 1]
```diff
@@ -36,14 +40,27 @@
...
+     * Referer link to track any redirect urls.
+     */
+    final String referer;
 
     InflightRequest(HttpServletRequest req) {
         String query = req.getQueryString();
         url = req.getRequestURL() + (query == null ? "" : "?" + query);
         startTime = System.currentTimeMillis();
+        userName = Jenkins.getAuthentication().getName();
+        referer = ((StaplerRequest)req).getReferer();
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/support/slowrequest/InflightRequest.java]: Unsafe cast. I would recommend to check the class and to default the field to something if it is not <code> StaplerRequest</code>

**Comment by stephenc** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
:+1:

**Pull request merged**